### PR TITLE
Fix for plugin timeout test

### PR DESCRIPTION
**Pull request by adityacb15**
Since the child:kill('sigkill') is unable to kill the child process (the sleep within the timeout.sh), I used [uv_kill](https://github.com/libuv/libuv/blob/v1.x/src/win/process.c#L1259) to achieve the result.

Also, here we need to prefix a minus (-) sign in front of the processID, since that will [kill all the processes](https://morningcoffee.io/killing-a-process-and-all-of-its-descendants.html) under this process group (i.e, killing all attached processes/ children)

### CODE REVIEW

File: check/base.lua [hunk 1]
```diff
@@ -427,14 +427,14 @@ function ChildCheck:_runChild(exePath, exeArgs, environ, callback)
 
   self._log(logging.DEBUG, fmt("%s: starting process", exePath))
 
-  local child = childprocess.spawn(exePath, exeArgs, { env = environ })
+  local child = childprocess.spawn(exePath, exeArgs, { env = environ, detached = true})
```
Comment (zzantozz) [inline check/base.lua:430]: Why change checks to spawn detached? This seems like it introduces a huge bug where when the agent is terminated, check processes will continue running, maybe even when the agent is shut down normally.

File: check/base.lua [hunk 2]
```diff
@@ -427,14 +427,14 @@ function ChildCheck:_runChild(exePath, exeArgs, environ, callback)
...
   child.stdin:destroy() -- close stdin for windows and ruby compatibility
 
   local pluginTimeout = timer.setTimeout(self._timeout, function()
     local timeoutSeconds = (self._timeout / 1000)
 
     self._log(logging.DEBUG, fmt("%s: Plugin didn't finish in %s seconds, killing it...", exePath, timeoutSeconds))
-    child:kill('sigkill')
+    uv.kill("-" .. child.pid, 9)
```
Comment (zzantozz) [inline check/base.lua:437]: This seems like it might be what was missing. Where did you find that it needs to be prefixed with a hyphen?

File: package.lua [hunk 3]
```diff
@@ -2,7 +2,7 @@ return {
   name = "rackspace-monitoring-agent",
   version = "2.6.23",
   luvi = {
-    version = "2.9.4-sigar",
+    version = "2.7.6-2-sigar",
```
Comment (zzantozz) [inline package.lua:5]: What does this have to do with this change? It's not mentioned in your PR or commit message.

**Comment by zzantozz** [comment]
Your PR description is good. The same message should be on your commit. PR's don't appear in git logs when browsing them. It's important to have complete messages on the commits.

File: check/base.lua (hunk 2 above)
Comment (adityacb15) [reply check/base.lua:437]: I have added the source article link in the PR description

File: package.lua (hunk 3 above)
Comment (adityacb15) [reply package.lua:5]: The correct sigar version is added to this PR, because the build was failing while looking for 2.9.4, a JAR for which doesn't exist

Comment (adityacb15) [reply package.lua:5]: I see an older open PR as well from Richa (correcting this exact same issue)

Comment (zzantozz) [reply package.lua:5]: Don't put unrelated changes in the same PR or commit. It should be it's own change, and if there's already a PR for the problem, let's get it merged.

File: check/base.lua (hunk 2 above)
Comment (zzantozz) [reply check/base.lua:437]: Oh I see. It's in the general process management one. I thought it was something specific to `uv`.
